### PR TITLE
Upgrade Hugo

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
```
site/docs/third-party-content/videos-and-presentations/index.html
  target does not exist --- site/docs/third-party-content/videos-and-presentations/index.html --> /css/prism.css
  target does not exist --- site/docs/third-party-content/videos-and-presentations/index.html --> /js/prism.js
```

I wonder why that is? Those files _should_ (in theory) be coming from the docsy theme 🤔

**Comment by Kalaiselvi84** [comment]
Targets with no Issue:
make site-server -> https://gist.github.com/Kalaiselvi84/3de02d537f998b79f74932abbf87321e
hugo server -> https://gist.github.com/Kalaiselvi84/2390cc082561579efed898f38ac9dc35
make site-static-preview -> https://gist.github.com/Kalaiselvi84/80e878043979fe0a910e65e2e30c6407

Issue:
Hugo test -> https://gist.github.com/Kalaiselvi84/0ccffae8405ba1bc7dfc37b54874e80f

**Comment by markmandel** [comment]
When you run `make site-server ` does the prism code syntax highlighting work for you locally?

**Comment by Kalaiselvi84** [comment]
> When you run `make site-server`  does the prism code syntax highlighting work for you locally?

both css/prism.css  and js/prism.css have 200 OK 
 https://screenshot.googleplex.com/4H6yJ2reVqYmPSx

https://screenshot.googleplex.com/7JwhtWmTW2WFR6B

Am I answering your question?

**Comment by markmandel** [comment]
You are!

If you look at the source code the page, is there more than one reference to a prism.js ?

**Comment by Kalaiselvi84** [comment]
I hope that you are referring to the HTML source code that is present in the Elements tab of the web page.

https://screenshot.googleplex.com/6oKRPcvTgpYunsN

Console tab says unable to load a source map file for the "asciinema-player.js" file
https://screenshot.googleplex.com/4w5iM5dxxeP68j9

**Comment by markmandel** [comment]
> I hope that you are referring to the HTML source code that is present in the Elements tab of the web page.

Yep! And clicking through to /js/prism.js - that all just works? 🤔 don't know what's up - that's odd it's failing on test.

**Comment by Kalaiselvi84** [comment]
> clicking through to /js/prism.js - that all just works?
yes, the prism.js code is there.
https://screenshot.googleplex.com/9ZYbWNCXbQXc9Dt

**Comment by markmandel** [comment]
The only weird idea I have is that `htmltest` doesn't like a `/from/root`  path like this when testing a folder? But that also seems weird.

Interesting test: On the old version of the website, is the html source for the prism files the same?

**Comment by Kalaiselvi84** [comment]
> On the old version of the website, is the html source for the prism files the same?

Yes, I have compared `js/prism.js` between the main branch and this Hugo feature branch, and there are no differences between these two branches

**Comment by markmandel** [comment]
> Yes, I have compared `js/prism.js` between the main branch and this Hugo feature branch, and there are no differences between these two branches

Not the files themselves, but the path to the files in the html source.

If I look at: `view-source:https://agones.dev/site/docs/installation/install-agones/yaml/ ` I see:

```html
<link rel="stylesheet" href="[/site/css/prism.css](https://agones.dev/site/css/prism.css)"/>
```

So it looks like it's the same.

For the test though we do generate to a local directory as a static copy of the website, so might be worth doing a compare and contrast of what is statically generated.

**Comment by Kalaiselvi84** [comment]
Difference between local and live websites:
Local (static copy): view-source:http://localhost:1313/docs/installation/install-agones/yaml/
Live: view-source:https://agones.dev/site/docs/installation/install-agones/yaml/

1.
Local (static copy): <head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
Live: <head>

2.
Local: <META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">
Live: <META NAME="ROBOTS" CONTENT="INDEX, FOLLOW">

3.
Local: NONE
Live: 
```
<script async src="https://www.googletagmanager.com/gtag/js?id=G-TK6J08ZNYX"></script>
<script>
var doNotTrack = false;
if (!doNotTrack) {
	window.dataLayer = window.dataLayer || [];
	function gtag(){dataLayer.push(arguments);}
	gtag('js', new Date());
	gtag('config', 'G-TK6J08ZNYX', { 'anonymize_ip': false });
}
</script>
```

4.
Local: <link rel="stylesheet" href="/css/prism.css"/>
Live: <link rel="stylesheet" href="/site/css/prism.css"/>

5.
Local: <link href="/scss/main.css" rel="stylesheet">
Live: <link rel="preload" href="/site/scss/main.min.d2ac6def1e694d73d9127adc8a08a66491f4a3d0a75fb93e754d56f4fc42811a.css" as="style">

6.
Local: NONE
Live: <link href="/site/scss/main.min.d2ac6def1e694d73d9127adc8a08a66491f4a3d0a75fb93e754d56f4fc42811a.css" rel="stylesheet" integrity="">


7.
Local: <script src='/js/prism.js'></script>
Live: <script src='/site/js/prism.js'></script>

Other then these differences, live website has "/site/" as part of the path.

**Comment by Kalaiselvi84** [comment]
words are disappearing😅. Pasted the differences in the buganizer ticket

**Comment by Kalaiselvi84** [comment]
@markmandel 

I've tried including different path configurations like 'css' or 'site,' or omitting these two, and I've used 'relURL' or 'absURL' for the `prism` reference. However, I'm still encountering a failure with the hugo-test target.

**Comment by markmandel** [comment]
Can you push one of the options to this draft PR, so I can have a look?

**Comment by Kalaiselvi84** [comment]
I included `/site/` and used `absURL` in this PR.
Hugo-test log -  https://gist.github.com/Kalaiselvi84/98e9280c3be7f052a834446a4c7386d6

File: site/themes/docsy/layouts/partials/head.html [hunk 1]
```diff
@@ -34,7 +33,7 @@
 {{end}}
 {{ if .Site.Params.prism_syntax_highlighting }}
 <!-- stylesheet for Prism -->
-<link rel="stylesheet" href="{{ "/css/prism.css" | relURL }}"/>
+<link rel="stylesheet" href="{{ "/site/prism.css" | absURL }}"/>
```
Comment (markmandel) [inline site/themes/docsy/layouts/partials/head.html:36]:
```suggestion
<script src='{{ "js/prism.js" | relURL }}'></script>
```

File: site/layouts/partials/head.html [hunk 2]
```diff
@@ -12,15 +12,14 @@
...
-{{- template "_internal/google_news.html" . -}}
 {{- template "_internal/schema.html" . -}}
 {{- template "_internal/twitter_cards.html" . -}}
 {{ if eq (getenv "HUGO_ENV") "production" }}
 {{ template "_internal/google_analytics.html" . }}
 {{ end }}
 {{ if .Site.Params.prism_syntax_highlighting }}
 <!-- stylesheet for Prism -->
-<link rel="stylesheet" href="{{ "/css/prism.css" | relURL }}"/>
+<link rel="stylesheet" href="{{ "/site/prism.css" | absURL }}"/>
```
Comment (markmandel) [inline site/layouts/partials/head.html:22]:
```suggestion
<link rel="stylesheet" href="{{ "css/prism.css" | relURL }}"/>
```

**Review by markmandel — COMMENTED** [review]
This doesn't solve all the issues (looks like there are a couple of other broken links with the version upgrade), but this solves the prism issues.

Approach taken from https://github.com/google/docsy/blob/5295589fd575084a17774a0e93a12f5708e9b7c4/layouts/partials/head.html#L41 (latest Docsy theme).

File: site/themes/docsy/layouts/partials/head.html (hunk 1 above)
Comment (Kalaiselvi84) [reply site/themes/docsy/layouts/partials/head.html]: Updated it. `hugo-test` failed with two errors. Details: https://gist.github.com/Kalaiselvi84/7f12221fa15b8e8707587039fcba59ec

**Comment by Kalaiselvi84** [comment]
Moved the 404.html file to the 'docsy' folder, and the error related to this file is resolved. Is it acceptable to keep the file in this location?

**Comment by Kalaiselvi84** [comment]
hugo-test is passed. here is the successful log - - https://gist.github.com/Kalaiselvi84/c044264e40b461a5b8db2a7a6f0ebbaf

**Comment by markmandel** [comment]
> Moved the 404.html file to the 'docsy' folder, and the error related to this file is resolved. Is it acceptable to keep the file in this location?

We don't actually use the 404, so I think we can delete it!

https://gohugo.io/templates/404/

This is what a 404 looks like for us: https://agones.dev/site/index3.html

**Comment by markmandel** [comment]
Cool cool - let's take a look at the preview when it comes up.

**Comment by Kalaiselvi84** [comment]
The hugo-test, site-server, site-static, and site-static-preview targets have successfully completed, with the total of two warnings.

**Comment by markmandel** [comment]
> The hugo-test, site-server, site-static, and site-static-preview targets have successfully completed, with the total of two warnings.

Sweet! I would like to give a look to see if anything's broken from a rendering perspective as well.

**Comment by Kalaiselvi84** [comment]
> Next stop 😄 let's try upgrading Docsy once this is merged.

Could you kindly advise on the most appropriate option for our project from [the link](https://www.docsy.dev/docs/updating/)?

**Comment by markmandel** [comment]
I know they don't seem to advocate for it, but let's start with a copy of docsy in our folder. I feel like we often need to make changes to the docsy theme, so having our own copy seems like a good idea.

**Pull request merged**